presto-spark-base/src/main/java/com/facebook/presto/spark/execution/common/FilePropertiesPopulator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.execution.common;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * An implementation of {@link PropertiesPopulator} that populates properties to a file on disk. The properties will be populated in the file with each line having <key>=<value>
+ * format
+ */
+public class FilePropertiesPopulator
```

Comment:
Right but I had another thought that generator generally means generating properties itself.
I think the workflow to be:

1. properties are generated from something external, to a list of properties in-memory. This stage if we are going to implement, shall be called property generator. Because we are generating properties to be used in the program.
2. a list of properties been materialized on disk. This is the stage we are implementing. Could either be materializer or populator. Because we are dumping in-memory properties to external.

Calling the 2). stage generator can be misleading in some scenarios.